components/camel-datasonnet/src/main/java/org/apache/camel/language/datasonnet/DatasonnetExpression.java

```diff
@@ -114,14 +114,20 @@ private Document<?> doEvaluate(Exchange exchange) {
         }
 
         Document<?> body;
+
         if (exchange.getMessage().getBody() instanceof Document) {
             body = (Document<?>) exchange.getMessage().getBody();
+        } else if (exchange.getMessage().getBody() == null || "".equals(MessageHelper.extractBodyAsString(exchange.getMessage()))) {
+            //Empty body, force type to be application/java
+            body = new DefaultDocument<>("", MediaTypes.APPLICATION_JAVA);
         } else if (MediaTypes.APPLICATION_JAVA.equalsTypeAndSubtype(bodyMT) || bodyMT == null) {
             body = new DefaultDocument<>(exchange.getMessage().getBody());
         } else {
             body = new DefaultDocument<>(MessageHelper.extractBodyAsString(exchange.getMessage()), bodyMT);
```

Comment:
You would essentially call this method twice, first when checking for empty, and then afterwards here too. I think it would be good to do this only once

Reply:
Makes sense, I refactored it.